### PR TITLE
Fix of [#38730], solarized colorscheme bug fix

### LINKED ISSUE #38730: Unreadable stacktrace in solarized theme
```julia
julia> versioninfo()
Julia Version 1.6.0-DEV.1537
Commit c36cf8cbc1 (2020-11-20 21:16 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.0 (ORCJIT, skylake)
Environment:
  JULIA_NUM_THREADS = 4
```
I have a terminal with solarized colorscheme, and in 1.6.0 stacktrace is unreadable.

flameshot-2020-12-06T09-14-16

I've tried to circumvent it with overriding `:light_black` color in `startup.jl`

```
atreplinit() do repl
    try
        Base.text_colors[:light_black] = Base.text_colors[241]
    catch
    end
end
```

But it helped only partially. Now I can see most of the messages except the number of the line where an error has happened.

flameshot-2020-12-06T09-09-13

### PR DESCRIPTION
**Pull request by Arkoniak**
Fixes #38730 
Add workaround of solarized scheme bug and expand `printstyled` functionality.

### CODE REVIEW

File: base/errorshow.jl [hunk 1]
```diff
@@ -735,22 +735,13 @@ function print_stackframe(io, i, frame::StackFrame, n::Int, digit_align_width, m
     pathparts = splitpath(file)
     folderparts = pathparts[1:end-1]
     if !isempty(folderparts)
-        printstyled(io, joinpath(folderparts...) * (Sys.iswindows() ? "\\" : "/"), color = :light_black)
+        printstyled(io, joinpath(folderparts...) * (Sys.iswindows() ? "\\" : "/"), color = :alt_light_black)
     end
 
-    # filename, separator, line
```
Comment (Keno) [inline base/errorshow.jl:741]: Dropped comment

File: base/client.jl [hunk 2]
```diff
@@ -30,6 +30,8 @@ answer_color() = text_colors[repl_color("JULIA_ANSWER_COLOR", default_color_answ
 stackframe_lineinfo_color() = repl_color("JULIA_STACKFRAME_LINEINFO_COLOR", :bold)
 stackframe_function_color() = repl_color("JULIA_STACKFRAME_FUNCTION_COLOR", :bold)
 
+alt_light_black_color() = repl_color("JULIA_ALT_LIGHT_BLACK_COLOR", :light_black)
```
Comment (Keno) [inline base/client.jl:33]: I'm not sure I like the "alt_light_black" name. Can we come up with a better name?

File: base/errorshow.jl (hunk 1 above)
Comment (Arkoniak) [reply base/errorshow.jl:741]: Fixed.

File: base/client.jl (hunk 2 above)
Comment (Arkoniak) [reply base/client.jl:33]: I definitely do not mind any other name, consider it as a placeholder.

Initially, I went with `:stackframe_color` but then decided that `:alt_light_black` is better.

**Comment by Arkoniak** [comment]
Sorry for bumping this up.

After I lived some time with this patch, it turns out that it doesn't solve all the problems.

1. `Pkg` also uses `:light_black` but in order to fix it, I need to change the code of some other package.
2. There is no guarantee that somebody somewhere is not going to use `:light_black` instead of `:alt_light_black` and it'll start over again. It will turn into an endless whack-a-mole fix, which is not good.

As an alternative solution, maybe substitute `:light_black` directly in `printstyled`? I.e. if we have an alternative color for `light_black` then use it instead of the original `light_black`? I do not quite like this approach because it messes with color names, but on the other hand, it'll work 100% of the time and will be applied only for in case when colorscheme is broken anyway.

**Comment by Arkoniak** [comment]
As an alternative, one can redefine `:light_black` directly in startup.jl, but I had issues with that, when I was running julia as a script from the prompt. It looks like `startup.jl` can be read after script is read, for example if script is malformed, than I get error messages in original invisible `:light_black`. It seems that `ENV` version is the only one that can guarantee that color is redefined before any other processing takes place.

**Comment by Keno** [comment]
I think a general colorscheme mechanism that lets people remap the named colors in the terminal would be fine. Shouldn't be just `light_black` at that point though.

**Comment by Arkoniak** [comment]
I'll never finish this PR and it'll never be accepted in it's current form, so there is no need to keep it open.

**Pull request closed without merging**